containerd-shim/process.go

```diff
@@ -186,23 +210,46 @@ func (p *process) create() error {
 		}
 		return err
 	}
-	p.stdio.stdout.Close()
-	p.stdio.stderr.Close()
+	if runtime.GOOS != "solaris" { // Temporary hack due to issues with pipes
+		p.stdio.stdout.Close()
+		p.stdio.stderr.Close()
+	}
 	if err := cmd.Wait(); err != nil {
 		if _, ok := err.(*exec.ExitError); ok {
 			return errRuntime
 		}
 		return err
 	}
-	data, err := ioutil.ReadFile("pid")
-	if err != nil {
-		return err
-	}
-	pid, err := strconv.Atoi(string(data))
-	if err != nil {
-		return err
+	if runtime.GOOS == "solaris" {
+		args := append([]string{
+			"--log", logPath,
+			"--log-format", "json",
+		}, p.state.RuntimeArgs...)
+		args = append(args, "start", p.id)
```

Comment:
Solaris won't support the split "create/start" behavior? This is basically making "create" the same as "run".

Is that only required to get the `pid`? Is there plan to add this support later on into `runz`?

It allows upper tools to correctly do some work before the container init process is actually started. On linux we currently use it to set the various monitoring fd.